Chrysalis-Papilio/Soen-490-Papilio
# [PAPILIO-17] User Profile Customization

# General info
Complete user profile customization (user bio, user password, user phone and user profile pic)

**Issue number**: PAPILIO-17

**Task description**: 
 - Added user profile customization for user bio, user pfp, phone number and password. 
- Improved the visuals of the user profile page
- General QoL improvements to the user profile code
- Further adherence to the MVVM pattern

# Testing
To test this feature you have to:

1. Clone the PAPILIO Backend Repo, then run the server by running the command "npm run dev" (also make sure the database url is in your .env file)
2. Open the android project in Android Studio and set up your chosen device (either emulator or actual phone connected to your PC)
3. Follow the steps in the Readme.MD found in the android project root folder
4. Start the app and create a new account
5. Navigate to the user profile page via the nav bar at the bottom
6. Press the "Edit Profile" and notice that some important UI elements pop up
7. Edit your bio
8. Press "Save Changes" and wait for your bio to be updated
9. Go back to the account menu page and come back to the user profile page and see if your new bio is still there. (Good if it is!)
10. Now press the "Edit Profile" button again and edit the user profile pic
11.  After the following the on-screen steps, verify that the user profile page now has your chosen picture as the pfp (Good if it does)
12. Repeat steps 10 and 11 but edit your password and phone number instead of the user pfp
13. Wait for success messages for both password and phone number editing 
14. Now log in as a Google account and repeat steps 5 onwards (You won't be able to edit passwords from Google accounts but everything else should be the same)

# Notes
- Unit tests and UI tests remaining (will be added tomorrow)
